ui/src/app/core/settings.service.ts

```diff
@@ -73,6 +84,16 @@ export class SettingsService {
     this.updateLocalStorage();
   }
 
+  setFilters(filters: Filter[], projectId: string): void {
+    this.currentSettings.v1.projects.forEach((p) => {
```

Comment:
tiny nit: I'd prefer to just use a regular loop and break rather than a `forEach` here, feels a bit more readable to me:
```
for (let p of this.currentSettings.v1.projects) {
  if (p.projectId === projectId) {
    p.filters = filters;
    break;
  }
}
```